app/models/concerns/foreman_resource_quota/host_managed_extensions.rb

```diff
@@ -113,5 +152,50 @@ def quota_assigment_optional?
     def call_utilization_helper(resources, hosts)
       utilization_from_resource_origins(resources, hosts)
     end
+
+    def add_host_capacity_to_quota(quota)
+      return if quota.nil?
+
+      update_quota_with_host_resources(quota) do |quota_resource_utilization, resource_value, _|
+        quota_resource_utilization + resource_value
+      end
+    end
+
+    def remove_host_capacity_from_quota(quota)
+      return if quota.nil?
+
+      update_quota_with_host_resources(quota) do |quota_resource_utilization, resource_value, resource_type|
+        quota_resource_utilization - helper_resource_value_subtraction(
+          quota.name,
+          resource_value,
+          quota_resource_utilization,
+          resource_type
+        )
+      end
+    end
+
+    def update_quota_with_host_resources(quota)
+      host_resources = determine_host_resources(quota.active_resources)
+      new_utilization = quota.utilization
+
+      host_resources.each do |resource_type, resource_value|
+        new_utilization[resource_type] ||= 0
+        new_utilization[resource_type] = yield(new_utilization[resource_type], resource_value, resource_type)
```

Comment:
Try to simplify the code such that it is easier to understand.

Reply:
Done.